argus/client/base.py

```diff
@@ -0,0 +1,170 @@
+import re
+from dataclasses import asdict
+from typing import Any, Type
+from uuid import UUID
+
+import requests
+
+from argus.backend.util.enums import TestStatus
+from argus.client.sct.types import LogLink
+
+JSON = dict[str, Any] | list[Any] | int | str | float | bool | Type[None]
+
+
+class ArgusClientError(Exception):
+    pass
+
+
+class ArgusClient:
+    schema_version: str | None = None
+
+    class Routes():
+        # pylint: disable=too-few-public-methods
+        SUBMIT = "/testrun/$type/submit"
+        HEARTBEAT = "/testrun/$type/$id/heartbeat"
+        SET_STATUS = "/testrun/$type/$id/set_status"
+        SET_PRODUCT_VERSION = "/testrun/$type/$id/update_product_version"
+        SUBMIT_LOGS = "/testrun/$type/$id/logs/submit"
+        FINALIZE = "/testrun/$type/$id/finalize"
+
+    def __init__(self, auth_token: str, base_url: str, api_version="v1") -> None:
+        self._auth_token = auth_token
+        self._base_url = base_url
+        self._api_ver = api_version
+
+    @property
+    def auth_token(self) -> str:
+        return self._auth_token
+
+    @staticmethod
+    def verify_location_params(endpoint: str, location_params: dict[str, str]) -> bool:
+        required_params: list[str] = re.findall(r"\$[\w_]+", endpoint)
+        for param in required_params:
+            if param.lstrip("$") not in location_params.keys():
+                raise ArgusClientError(f"Missing required location argument for endpoint {endpoint}: {param}")
+
+        return True
+
+    @staticmethod
+    def check_response(response: requests.Response, expected_code: int = 200) -> JSON:
```

Comment:
I'd argue if `check_response` should return anything - name rather indicates verification, not processing any data.

Reply:
Would `parse_response` be better name? That's one I used previously for this method, but then switched to this one.

Reply:
no, `check_response` name is ok, but should not return/process anything (here by process I mean `response_data: JSON = response.json()`). Generally, if name indicates only verification let's not add another purpose to method (single responsibility).